src/api/login/login.ts

```diff
@@ -74,30 +85,40 @@ router.post('/', checkSchema({
         email: 'lucas@gmail.com'
     }
     const errors = validationResult(req);
-
-    if (!errors.isEmpty()) {
-        return res.status(422).json({ errors: errors.array() });
-    }
-    
-    // Mock database request for user login
-    if(req.body.user == "lucas@gmail.com" && req.body.password == 'safepass'){
-        jwt.sign({user: user}, 'secretKeyHere', { expiresIn: '30s' } ,(err, token) => {
-            res.json({
-                token: token
+    try {
+        if (!errors.isEmpty()) {
+            throw new ValidationError(errors.array());
+        }
+        
+        // Mock database request for user login
+        if(req.body.user == "lucas@gmail.com" && req.body.password == 'safepass'){
+            jwt.sign({user: user}, 'secretKeyHere', { expiresIn: '30s' } ,(err, token) => {
+                res.json({
+                    token: token
+                });
             });
-        });
-    } else {
-        res.send('Your credentials are invalid');
+        } else {
+            throw new BadRequestError({
+                error: "Your credentials are invalid"
+              });
+        }
+    } catch(e) {
+        res.send(e)
     }
 });
 
 function verifyToken(req, res, next) {
     const authToken = req.headers['authorization'];
-    if(typeof authToken !== 'undefined') {
-        req.token= authToken;
-        next();
-    } else {
-        res.sendStatus(403);
+
+    try {
+        if(typeof authToken !== 'undefined') {
+            req.token= authToken;
+            next();
+        } else {
+            throw new BadRequestError('Not allowed.');
+        }        
+    } catch(e) {
+        res.send(e)
```

Comment:
never throw error to the enduser.
no need for try/catch blocks;

The happy path call `next()` and the worst case throw error
here you could use `UnauthorizedError()`;